crypto/src/core/rlwe/test.rs

```diff
@@ -466,14 +498,16 @@ fn noise_stats() {
         }
 
         let ring_size = param.ring_size as f64;
-        let var_sk = param.sk_distribution.variance();
         let var_noise = param.noise_distribution.variance();
-        let var_noise_ct_pk =
-            ring_size * var_sk * var_noise + (ring_size * var_sk + 1.0) * var_noise;
+        let var_u = param.u_distribution.variance();
+        let var_sk = param.sk_distribution.variance();
+        let var_noise_pk = var_noise;
+        let var_noise_ct_sk = var_noise;
+        let var_noise_ct_pk = ring_size * (var_u * var_noise_pk + var_sk * var_noise) + var_noise;
```

Comment:
I'm not sure whether float64 is high enough precision for this calculation. 

A better idea will be to generate these tests using a sage script.